Made scheme, input_format, and build_directory arguments optional to …

taiheng opened the pull request:
…match slather requirements.

This fixes krausefx/fastlane#650.

xfreebird committed:
Improved unlock_keychain action

taiheng committed:
Made scheme, input_format, and build_directory arguments optional to match slather requirements.

KrauseFx commented:
Looks good, thanks for the pull request. Can you also update the tests so that they don't fail? You can run them using `rspec`. More information in the [countdown repo](https://github.com/fastlane/countdown).

I can also take over from here if you prefer.

KrauseFx committed:
Merge pull request #672 from xfreebird/feature/unlock_keychain_improvement

Improved unlock_keychain action

taiheng commented:
I'll take a look at the tests, sorry about that.

KrauseFx commented:
Cool, no problem. Let me know if it doesn't work and I should take over migrating the tests.

Commit:
Changed name of slather parameter from proj to xcodeproj to better match default action naming and slather documentation.
Updated tests of slather action to only require xcodeproj parameter, all other parameters are now optional in line with current released version of slather (v1.8.1). Previously build_directory, scheme, and proj were required and would cause an error when not provided.

Commit:
Merge branch 'master' of https://github.com/tucali/fastlane

Commit:
Merge branch 'tucali/master'

taiheng commented:
I've updated the tests to reflect that the only parameter required is the project name, and also changed the name of this parameter to better match the default actions in fastlane from 'proj' to 'xcodeproj'. This parameter name also seems out of place with the other parameters in this action and is the only one that is truncated for some reason.

KrauseFx commented on lib/fastlane/actions/slather.rb:
```diff
@@ -49,22 +49,23 @@ def self.available_options
-                                       env_name: "FL_SLATHER_PROJ", # The name of the environment variable
+                                       optional: true
+                                      ),
+          FastlaneCore::ConfigItem.new(key: :xcodeproj,
```
Sorry, we can't just change the key of this option, as it would break existing setups. Please revert this change to be `:proj` again.

Commit:
Revert key change in slather from xcodeproj back to proj.

cyupa commented:
Any chance you'll merge this, @KrauseFx?

taiheng commented:
Reverted breaking change and test.

KrauseFx commented on lib/fastlane/actions/slather.rb:
```diff
@@ -49,22 +49,23 @@ def self.available_options
+                                      ),
           FastlaneCore::ConfigItem.new(key: :proj,
-                                       env_name: "FL_SLATHER_PROJ", # The name of the environment variable
+                                       env_name: "FL_SLATHER_PROJECT", # The name of the environment variable
```
:cry:

KrauseFx commented:
Thanks @taiheng
